src/policy/space.rs

```diff
@@ -292,11 +318,25 @@ impl<'a> SFTMap<'a> {
         self_mut.sft[chunk] = sft;
     }
 
-    pub fn is_in_space(&self, object: ObjectReference) -> bool {
+    pub fn is_in_any_space(&self, object: ObjectReference) -> bool {
         if object.to_address().chunk_index() >= self.sft.len() {
             return false;
         }
-        self.get(object.to_address()).is_mmtk_object(object)
+        self.get(object.to_address()).is_in_space(object)
+    }
+
+    #[cfg(feature = "global_alloc_bit")]
+    pub fn is_mmtk_object(&self, addr: Address) -> bool {
+        if addr.is_zero() {
+            return false;
+        }
+        if !addr.is_aligned_to(1usize << alloc_bit::ALLOC_SIDE_METADATA_SPEC.log_bytes_in_region) {
```

Comment:
We need to be precise here.  Returning `true` means the GC will attempt to trace through it. It will try to access object fields using `addr.to_object_reference()` as the object reference.  If `addr` is slightly misaligned, the GC will end up reading fields at wrong offsets, too.

I'll change the code and take the offset into consideration.